cstrml/benefits/src/lib.rs

```diff
@@ -0,0 +1,262 @@
+    }
+
+    fn maybe_reduce_fee(who: &<T as frame_system::Config>::AccountId, fee: BalanceOf<T>, reasons: WithdrawReasons) -> Result<NegativeImbalanceOf<T>, DispatchError> {
+        Self::maybe_reduce_fee_inner(who, fee, reasons)
+    }
+
+    fn maybe_free_count(who: &<T as frame_system::Config>::AccountId) -> bool {
+        Self::maybe_free_count_inner(who)
+    }
+}
+
+
+decl_storage! {
+    trait Store for Module<T: Config> as Benefits {
+        // Overall benefits information
+        OverallBenefits get(fn overall_benefits): EraBenefits<BalanceOf<T>>;
+        // One fee reduction information
+        FeeReductionLedger get(fn fee_reduction_ledger): map hasher(blake2_128_concat) T::AccountId => FeeReductionBenefit<BalanceOf<T>>;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        #[weight = 1000]
+        pub fn add_benifit_funds(origin, #[compact] value: BalanceOf<T>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            // 1. Reserve the currency
+            T::Currency::reserve(&who, value.clone()).map_err(|_| Error::<T>::InsuffientBalance)?;
+
+            // 2. Upgrade collateral.
+            <FeeReductionLedger<T>>::mutate(&who, |fee_reduction| {
+                    fee_reduction.funds += value.clone();
+                    fee_reduction.total_fee_reduction_count = Self::calculate_total_count_reduction(&fee_reduction.funds);
+                }
+            );
+            <OverallBenefits<T>>::mutate(|benefits| { benefits.total_funds += value.clone();});
+
+            // 3. Emit success
+            Self::deposit_event(RawEvent::AddBenefitFundsSuccess(who.clone(), value));
+
+            Ok(())
+        }
+
+        #[weight = 1000]
+        pub fn cut_benifit_funds(origin, #[compact] value: BalanceOf<T>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            // 1. Unreserve the currency
+            T::Currency::unreserve(&who, value.clone());
+
+            // 2. Upgrade collateral.
+            <FeeReductionLedger<T>>::mutate(&who, |fee_reduction| {
+                    fee_reduction.funds = fee_reduction.funds.saturating_sub(value.clone());
+                    fee_reduction.total_fee_reduction_count = Self::calculate_total_count_reduction(&fee_reduction.funds);
+                }
+            );
+            <OverallBenefits<T>>::mutate(|benefits| { benefits.total_funds = benefits.total_funds.saturating_sub(value.clone());});
+
+            // 3. Emit success
+            Self::deposit_event(RawEvent::CutBenefitFundsSuccess(who.clone(), value));
+
+            Ok(())
+        }
+    }
+}
+
+
+impl<T: Config> Module<T> {
+    /// The return value is the used fee quota in the last era
+    pub fn update_era_benefit_inner(next_era: EraIndex, total_benefits: BalanceOf<T>) -> BalanceOf<T> {
```

Comment:
change this. keep the `maybe_reduce_fee_inner` and `maybe_free_count` since `do_maybe_reduce_fee_inner` or `maybe_do_reduce_fee_inner` looks strange....